mobile/lib/ui/viewer/file/file_widget.dart

```diff
@@ -40,16 +40,8 @@ class FileWidget extends StatelessWidget {
         key: key ?? ValueKey(fileKey),
       );
     } else if (file.fileType == FileType.video) {
-      // use old video widget on iOS simulator as the new one crashes while
-      // playing certain videos on iOS simulator
-      // if (kDebugMode && Platform.isIOS) {
-      //   return VideoWidget(
-      //     file,
-      //     tagPrefix: tagPrefix,
-      //     playbackCallback: playbackCallback,
-      //   );
-      // }
-      return VideoWidgetNative(
+      return VideoViewWidget(
```

Comment:
We should still use the native player for non-preview videos.
Also, let's add the logic here to check if it's a internal User and the video has preview, then use the PreviewVideoWidget otherwise stick to existing logic. Also, if the video is not uploaded, we just stick to existing logic here.

Inside FileDataService, we can cache the fileIDs for all videos that have preview (that cache can be populcated in sync manner during app startup).

Reply:
Resolved!